tests/unit/flow-construct/test_flow.py

```diff
@@ -827,3 +827,13 @@ def test_connect_to_predecessor():
     assert len(f._pod_nodes['gateway'].head_args.hosts_in_connect) == 0
     assert len(f._pod_nodes['pod1'].head_args.hosts_in_connect) == 0
     assert len(f._pod_nodes['pod2'].head_args.hosts_in_connect) == 1
+
+
+def test_flow_grpc_with_shard():
+    with pytest.raises(
+        NotImplementedError, match='GRPC data runtime does not support sharding'
+    ):
+        Flow(grpc_data_requests=True).add(shards=2)
+        Flow(grpc_data_requests=True).add(shards=None)
+
+    Flow(grpc_data_requests=False).add(shards=-1)
```

Comment:
Let's use the good case here which is 1. Also I would also add a good case without any shard like `Flow(grpc_data_requests=False).add()`
I believe the no shard case will fail with your change, because kwargs does not contain 'shards' then and you might get a KeyError.

```suggestion
    Flow(grpc_data_requests=False).add(shards=1)
```

Reply:
With condition:
```python
if args.grpc_data_requests and kwargs.get('shards') is not None and kwargs.get('shards') > 1:
            raise NotImplementedError("GRPC data runtime does not support sharding")
```
I ran `Flow(grpc_data_requests=False).add()` case and it didnt gave any error.
```python
>>> Flow(grpc_data_requests=False).add()
<jina.flow.base.Flow object at 0x7f1c765636a0>
```